src/shared/extensionUtilities.ts

```diff
@@ -374,37 +375,60 @@ export class ExtensionUserActivity {
     /** The event that is triggered when the user interacts with the extension */
     onUserActivity = this.activityEvent.event
 
-    /** To be efficient we re-used instances that already have the same throttle delay */
-    private static instances: { [delay: number]: ExtensionUserActivity } = {}
-
     /**
      * @param delay The delay until a throttled user activity event is fired in milliseconds.
-     * @param activityEvents The events that trigger the user activity.
+     * @param activityEvents For testing purposes. The events that trigger the user activity.
      * @returns
      */
-    static instance(delay: number = 500, activityEvents = ExtensionUserActivity.activityEvents): ExtensionUserActivity {
-        if (!ExtensionUserActivity.instances[delay]) {
-            const newInstance = new ExtensionUserActivity(delay, activityEvents)
-            ExtensionUserActivity.instances[delay] = newInstance
-        }
-        return ExtensionUserActivity.instances[delay]
-    }
+    constructor(delay: number = 10_000, activityEvents = ExtensionUserActivity.activityEvents) {
+        // This ensures we don't fire a user activity event more than once per delay.
+        const throttledEvent = _.throttle(
+            (event, obj) => {
+                if (event === vscode.window.onDidChangeWindowState && obj.active === false) {
+                    // Edge case where this is randomly emitted without any user interaction.
+                    return
+                }
+                if (event === vscode.workspace.onDidChangeTextDocument) {
+                    // Our extension may change a file in the background (eg: log file)
+                    const uriOfDocumentUserIsEditing = vscode.window.activeTextEditor?.document?.uri
+                    if (!uriOfDocumentUserIsEditing) {
+                        // User was not editing any file themselves
+                        return
+                    }
+                    if (uriOfDocumentUserIsEditing !== (obj as vscode.TextDocumentChangeEvent).document.uri) {
+                        // File changed was not the one the user was actively working on
+                        return
+                    }
+                }
+                this.activityEvent.fire()
```

Comment:
Do we have a log message for this? We need a log message like:

    ExtensionUserActivity: event "${eventName}" 

so that we can troubleshoot in case we have "spurious activity", other than the special-cases already checked above.